packages/components/src/components/rating-stars/rating-stars.tsx

```diff
@@ -9,298 +9,195 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
  */
 
-/*
-adapted from shoelace's rating component
-https://github.com/shoelace-style/shoelace/blob/next/src/components/rating/rating.ts
-*/
-
 import {
   Component,
   h,
   Prop,
   Host,
   Element,
-  State,
   Event,
   EventEmitter,
+  State,
 } from '@stencil/core';
-import { clamp, handleListeners } from './utils/utils';
-import classNames from 'classnames';
-import statusNote from '../../utils/status-note';
+
+export interface StarInterface extends HTMLDivElement {
+  dataset: {
+    value: string;
+    selected?: string;
+  };
+}
+const sizes = {
+  small: 16,
+  large: 24,
+};
+let ratingStarCount = 0;
 
 @Component({
   tag: 'scale-rating-stars',
   styleUrl: 'rating-stars.css',
   shadow: true,
 })
 export class RatingStars {
-  element: HTMLElement;
-
-  @Element() hostElement: HTMLElement;
-  /** (optional) hoverValue  */
-  @State() hoverValue = 0;
-  /** (optional) hoverValue  */
-  @State() isHovering = false;
-  /** (optional) max  */
-  @Prop() max = 5;
-  /** (optional) value  */
-  @Prop({ mutable: true }) value = 0;
-  /** (optional) small  */
-  @Prop() small = false;
-  /** (optional) disabled  */
-  @Prop() disabled = false;
-  /** (optional) ariaLabelTranslation  */
-  @Prop({ mutable: true })
-  ariaLabelTranslation = `${this.value} out of ${this.max} stars`;
-  /** (optional) precision  */
-  @Prop() precision = 1;
-  /** (optional) slider label */
-  @Prop() label?: string;
-  /** Emitted when the value has changed. */
-  @Event() scaleRatingChange!: EventEmitter<{ value: number }>;
-
-  colorFilled = `var(--scl-color-primary)`;
-  colorBlank = `var(--scl-color-grey-50)`;
-  size = this.small ? '16px' : '24px';
+  @Element() host: HTMLElement;
+
+  ratingStarId = `scale-rating-star-${ratingStarCount++}`;
+
+  /** The lower limit of the rating. In cases where  */
+  @Prop({ reflect: true }) starSize: 'small' | 'large' = 'large';
+  /** The lower limit of the rating. In cases where  */
+  @Prop({ reflect: true }) minRating = 0;
+  /** The upper limit of the rating */
+  @Prop({ reflect: true }) maxRating = 5;
+  /** Represents the current value of the rating */
+  @Prop({ mutable: true, reflect: true }) rating = 0;
```

Comment:
I wonder what happened to this prop being called `value`. The reason for having `value` is consistency with the rest of the "form control" components (and form controls in general, it's like a convention) and also playing nice with frameworks (like Angular) that expect the property holding the value to be called `value` in their binding mechanisms.

I'm not asking now that you change it back, I just would like to know the reasoning behind the change :)

Reply:
We've discussed this with @oddcelot, and we don't agree haha but it's clear 👍